### PR TITLE
Fix/updates

### PR DESCRIPTION
**Pull request by woile**
- Some refactors to make the code clearer
- Added `partition`, `offset` and `message.id` to the send span

### CODE REVIEW

File: src/opentelemetry_instrumentation_kstreams/utils.py [hunk 1]
```diff
@@ -83,10 +89,20 @@ def extract_send_topic(args, kwargs) -> Any:
...
-        return KStreamsKafkaExtractor._extract_argument(
-            "partition", 4, None, args, kwargs
-        )
+    def extract_send_partition(record_metadata: Any) -> Any:
+        if hasattr(record_metadata, "partition"):
+            return record_metadata.partition
+
+    @staticmethod
+    def extract_send_offset(record_metadata: Any) -> Any:
+        if hasattr(record_metadata, "offset"):
```
Comment (marcosschroh) [inline src/opentelemetry_instrumentation_kstreams/utils.py]: You can do `return getattr(record_metadata, "offset", None)`, I think the return type should be `Optional[int]`. Similar for the method `extract_consumer_group`

File: src/opentelemetry_instrumentation_kstreams/utils.py [hunk 2]
```diff
@@ -83,10 +89,20 @@ def extract_send_topic(args, kwargs) -> Any:
         )
 
     @staticmethod
-    def extract_send_partition(args: Any, kwargs: Any) -> Any:
-        return KStreamsKafkaExtractor._extract_argument(
-            "partition", 4, None, args, kwargs
-        )
+    def extract_send_partition(record_metadata: Any) -> Any:
```
Comment (marcosschroh) [inline src/opentelemetry_instrumentation_kstreams/utils.py]: I think the return type should be `int`

File: src/opentelemetry_instrumentation_kstreams/utils.py [hunk 3]
```diff
@@ -101,57 +117,101 @@ def extract_consumer_client_id(instance: Stream) -> str:
...
 
 
-def _get_span_name(operation: str, topic: str):
+def _get_span_name(operation: str, topic: str) -> str:
     return f"{topic} {operation}"
 
 
 def _wrap_send(tracer: Tracer) -> Callable:
-    async def _traced_send(func, instance: StreamEngine, args, kwargs):
+    async def _traced_send(func, instance: StreamEngine, args, kwargs) -> Any:
```
Comment (marcosschroh) [inline src/opentelemetry_instrumentation_kstreams/utils.py]: I think the return type should be `typing.Awaitable[RecordMetadata]`. The argument `func` should be typed as func: [Send](https://github.com/kpn/kstreams/blob/master/kstreams/types.py#L15) if I am not mistaken

**Review by marcosschroh — APPROVED** [review]

File: src/opentelemetry_instrumentation_kstreams/utils.py (hunk 3 above)
Comment (woile) [reply src/opentelemetry_instrumentation_kstreams/utils.py]: thanks!!

**Review by marcosschroh — APPROVED** [review]

**Pull request merged**